src/sdam/topology.ts

```diff
@@ -628,6 +647,11 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
           );
         }
 
+        if (options.timeout) {
+          throw new MongoOperationTimeoutError('Timed out during server selection', {
```

Comment:
Should we include the TopologyDescription in this error (regular server selection timeout errors do).

Reply:
Is adding the originally thrown error as a cause not sufficient for this?